01-data-structures/03-linked-lists/linked_list_spec.rb

```diff
@@ -89,4 +89,41 @@
       expect(llist.head).to eq nil
     end
   end
-end
\ No newline at end of file
+
+  describe "#locate_node_by_index" do
+    before do
+      llist.add_to_tail(n1)
+      llist.add_to_tail(n2)
+      llist.add_to_tail(n3)
+    end
+
+    it "locates nodes by index" do
+      expect(llist.locate_node_by_index(0).data).to eq "Rob"
+      expect(llist.locate_node_by_index(1).data).to eq "Ben"
+      expect(llist.locate_node_by_index(2).data).to eq "Mike"
+      llist.delete(n2)
+      expect(llist.locate_node_by_index(0).data).to eq "Rob"
+      expect(llist.locate_node_by_index(1).data).to eq "Mike"
+    end
+
+    describe "#delete_node_by_index" do
```

Comment:
you should try where the index is greater than the list length or the index is an invalid argument